sound/soc/sof/ops.h

```diff
@@ -330,6 +330,35 @@ snd_sof_pcm_platform_trigger(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+/* host read DSP device data */
+static inline void snd_sof_ipc_read(struct snd_sof_dev *sdev,
+					void *p, size_t sz)
+{
+	if (sof_ops(sdev) && sof_ops(sdev)->ipc_read)
+		sof_ops(sdev)->ipc_read(sdev, p, sz);
+}
+
+/* host read DSP stream data */
+static inline void snd_sof_ipc_read_pcm_stream(struct snd_sof_dev *sdev,
+					   struct snd_pcm_substream *substream,
+					   void *p, size_t sz)
+{
+	if (sof_ops(sdev) && sof_ops(sdev)->ipc_read_pcm_stream)
+		sof_ops(sdev)->ipc_read_pcm_stream(sdev, substream, p, sz);
+}
+
+/* host configure DSP HW parameters */
+static inline int snd_sof_ipc_read_pcm_params(struct snd_sof_dev *sdev,
+				struct snd_pcm_substream *substream,
+				const struct sof_ipc_pcm_params_reply *reply)
+{
+	if (sof_ops(sdev) && sof_ops(sdev)->ipc_read_pcm_params)
+		return sof_ops(sdev)->ipc_read_pcm_params(sdev, substream,
+							  reply);
+
+	return 0;
```

Comment:
isn't this mandatory and is 0 a relevant return value?

Reply:
I'm not sure, if you think it should be mandatory, we can enforce it, sure